bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/execution/event/EventsHandler.java

```diff
@@ -367,8 +368,12 @@ private void triggerCatchStartEventSubProcess(final SEventTriggerType triggerTyp
                     processExecutor, logger);
             interruptor.interruptProcessInstance(parentProcessInstanceId, SStateCategory.ABORTING, -1, subProcflowNodeInstance.getId());
         }
-        processExecutor.start(processDefinitionId, targetSFlowNodeDefinitionId, 0, 0, operations.getContext(), operations.getOperations(),
-                null, null, subProcflowNodeInstance.getId(), subProcessId, null); // Should we support process contract inputs on EventSubProcess?
+        try {
+            processExecutor.start(processDefinitionId, targetSFlowNodeDefinitionId, 0, 0, operations.getContext(), operations.getOperations(), null, null,
+                    subProcflowNodeInstance.getId(), subProcessId, null); // Process contract inputs on EventSubProcess are not supported.
+        } catch (ContractViolationException e) {
+            throw new SProcessInstanceCreationException("Process contracts are not allowed on Event sub process");
```

Comment:
Do we have to do that? it will never happen because builders do not add it in the definition , maybe the exception before the here should be a server exception and not a client exception